src/recompilation.cpp

```diff
@@ -511,17 +526,34 @@ bool process_instruction(const RecompPort::Context& context, const RecompPort::C
         {
             uint32_t branch_target = instr.getBranchVramGeneric();
             if (branch_target == instr_vram) {
-                print_line("void pause_self(uint8_t *rdram); pause_self(rdram)");
+                print_line("pause_self(rdram)");
             }
             // Check if the branch is within this function
             else if (branch_target >= func.vram && branch_target < func_vram_end) {
                 print_unconditional_branch("goto L_{:08X}", branch_target);
             }
             // Otherwise, check if it's a tail call
             else if (instr_vram == func_vram_end - 2 * sizeof(func.words[0])) {
```

Comment:
I don't think this special handling is needed anymore, since we know any j or b that points to another function can be treated as a tail call regardless of where it is in the caller function. The else if you added should cover every case this one covers